devicetypes/smartthings/zigbee-multi-button.src/zigbee-multi-button.groovy

```diff
@@ -223,3 +240,45 @@ private getModelBindings(model) {
 	}
 	bindings
 }
+
+private Map parseAduroSmartButtonMessage(Map descMap){
+	def buttonState = "pushed"
+	def buttonNumber = 0
+	if (descMap.clusterInt == zigbee.ONOFF_CLUSTER) {
+		if (descMap.command == "01") {
+		    buttonNumber = 1
+		}else if (descMap.command == "00") {
+		    buttonNumber = 4
+		}else {
+		    return [:]
+		}
+	}else if (descMap.clusterInt == zigbee.LEVEL_CONTROL_CLUSTER) {
+		if (descMap.command == "02") {
+		    def data = descMap.data
+		    def d0 = data[0]
+		    if (d0 == "00") {
+			buttonNumber = 2
+		    }
+		    if (d0 == "01") {
+			buttonNumber = 3
+		    }
+		}
+	}else if (descMap.clusterInt == ADUROSMART_SPECIFIC_CLUSTER) {
+		def list2 = descMap.data
+		buttonNumber = (list2[1] as int) + 1
+	}
+	if (buttonNumber !=0) {
+		def childevent = createEvent(name: "button", value: "pushed",data: [buttonNumber: 1], isStateChange: true)
+		sendEventToChild(buttonNumber, childevent)
+		def descriptionText = "$device.displayName button $buttonNumber was $buttonState"
+		return createEvent(name: "button", value: buttonState, data: [buttonNumber: buttonNumber], descriptionText: descriptionText, isStateChange: true)
+        }else {
```

Comment:
nit: spaces instead of tabs again

your editor should have a function that allows you to convert spaces->tabs all at once